### ISSUE TITLE
Tags / branches out of date?

### ISSUE THREAD

**Issue by wjakob** [RLovelett/eigen#3]
Hi Ryan,

thank you very much for keeping this mirror, it's very helpful (e.g.  via git submodule).
I have encountered one issue though, which is that the tags and branches are stale (the last release tag was from Mar 7, 2013). Is it possible to keep the tags and branches in sync? This way, it will be much easier to reference a specific version from your repository.

Thanks,
Wenzel

**Comment by RLovelett** [comment]
I won't promise anything but I'll certainly look into it. Frankly I don't know much about Mercurial (which is what Eigen uses). So if you know any articles/commands for me to look at it'd be appreciated.

**Comment by wjakob** [comment]
Hi Ryan,

I don't have enough details on how you keep the repositories on sync. Do you use a particular software or script?

Thanks,
Wenzel

**Comment by RLovelett** [comment]
@wjakob that seems like a reasonable request.

It is nothing too crazy. It is just a Jenkins build script that runs every day. The first step is it clones (checks out, not sure the correct Mercurial term here) this repository: https://bitbucket.org/eigen/eigen.

Then in that directory runs this script.

```
hg bookmark -r default master
hg push git+ssh://git@github.com/RLovelett/eigen.git
hg push
```

It does make use of [hg-git](http://hg-git.github.io/) to make everything work. Can't remember which version I use and if I just need a newer version of it I have no problem updating.

**Comment by jansendup** [comment]
The missing tags is due to only pushing the default branch in the mercurial repository. The missing tags are on the other branches. Would it be possible for you to mirror all the branches?

**Comment by RLovelett** [comment]
@jansendup as I've said before its totally possible. I've provided the script that runs (see my 13 Aug 2015 comment) provide a modification to it that mirrors all the branches and I'd be happy to update it.

As it stands right now I'm not interested in spending my personal time to resolve it because this works just fine for my needs as is. I don't mean that to be flippant just to say I don't know how; otherwise I'd do it.

**Comment by RLovelett** [comment]
@jeffmelville no I do not do a fresh clone of the mercurial repo each time.

I tried running those commands and when I got to `hg push ../eigen-git` mercurial spit out:

```
pushing to eigen-git/
searching for changes
no changes found
```

Granted I did not clone into a new bare _Git_ repository. I used my local clone of this repository. Other than that it was the same.

Also I did some research into the branch mirroring thing. I doubt that'll be as straight forwards as I'd ever hope. From reading the docs over at hg-git it seems that Mercurial's branching model is entirely different from Git's model. Thus the need for bookmarks.

From what I can tell I'd probably have to write a script to loop over the Mercurial branches and create bookmarks and then push for each. Not really something I want to write or maintain.

With all of that in mind. If anyone wants push access to the repository to add branch mirrors I'd be happy to grant it.
